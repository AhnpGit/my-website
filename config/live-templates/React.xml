<templateSet group="React">
  <template name="rx_module" value="const _ACTION_PREFIX = '$projectName$/$moduleName$'&#10;&#10;const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;&#10;// TODO: Connect the reducer.&#10;export default function reducer(state = {$defaultState$}, action) {&#10;    switch (action.type) {&#10;        case $ACTION_NAME$:&#10;            return {$END$}&#10;&#10;        default:&#10;            return state&#10;    }&#10;}&#10;&#10;export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$payload$},&#10;    meta: {},&#10;    error: undefined,&#10;})&#10;" description="Creates a DUCKS-module" toReformat="true" toShortenFQNames="true">
    <variable name="projectName" expression="" defaultValue="&quot;&lt;project name&gt;&quot;" alwaysStopAt="true" />
    <variable name="moduleName" expression="fileNameWithoutExtension()" defaultValue="" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <variable name="defaultState" expression="" defaultValue="" alwaysStopAt="true" />
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="payload" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component" value="import React from 'react'&#10;&#10;const $ComponentName$ = ({$props$}) =&gt; (&#10;    &lt;$END$/&gt;&#10;)&#10;&#10;export default $ComponentName$" description="Creates a functional component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="props" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component_class" value="import React from 'react'&#10;&#10;class $ComponentName$ extends React.Component {&#10;    render() {&#10;        return (&#10;            &lt;$END$/&gt;&#10;        )&#10;    }&#10;}&#10;&#10;export default $ComponentName$" description="Creates a class component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_proptypes" value="import PropTypes from 'prop-types'&#10;$ComponentName$.propTypes = {&#10;    $prop$: PropTypes.$END$,&#10;}" description="Adds proptypes to a component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="prop" expression="" defaultValue="&quot;prop&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action_async" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;const $ACTION_NAME$_PENDING = `${$ACTION_NAME$}_PENDING`&#10;const $ACTION_NAME$_FULFILLED = `${$ACTION_NAME$}_FULFILLED`&#10;const $ACTION_NAME$_REJECTED = `${$ACTION_NAME$}_REJECTED`" description="Creates async action types" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`" description="Creates an action type" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$END$},&#10;    meta: {},&#10;    error: undefined,&#10;})" description="Creates an action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator_async" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: getPayload(),&#10;    meta: {},&#10;    error: undefined,&#10;    &#10;    async function getPayload() {&#10;        return $END$&#10;    }&#10;})" description="Creates an async action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_container" value="import { connect } from 'react-redux'&#10;import $ComponentName$ from '../components/$ComponentName$'&#10;&#10;const mapStateToProps = (state) =&gt; ({&#10;    $END$&#10;})&#10;&#10;const mapDispatchToProps = (dispatch) =&gt; ({&#10;})&#10;&#10;export default connect(&#10;    mapStateToProps,&#10;    mapDispatchToProps,&#10;)($ComponentName$)" description="Creates a container class" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_styled" value="import styled from 'styled-components'&#10;&#10;const $ComponentName$ = styled.$ElementType$`&#10;  $END$&#10;`&#10;&#10;export default $ComponentName$" description="Creates a styled component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="ElementType" expression="" defaultValue="&quot;div&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
</templateSet>
