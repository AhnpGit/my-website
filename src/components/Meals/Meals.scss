.wrapper__meals {
  lost-center: 100% 0 flex;
  .meal__photo {
    lost-row: 1/4 0 0;
    overflow: hidden;
    background-color: $black;
  }

  @include media-query($medium-up) {
    padding: 0 $base-unit;
    .meal__photo {
      lost-column: 1/4 0 0;
    }
  }

  img {
    opacity: 0.7;
    transform: scale(1.18);
    transition: transform 0.5s, opacity 0.5s;

    &:hover {
      opacity: 1;
      transform: scale(1.08);
    }
  }
}
