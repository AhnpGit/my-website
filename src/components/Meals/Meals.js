// @flow

import React from 'react'

import type { MealsType } from '../../utils/types'
import CustomPropTypes from '../../utils/CustomPropTypes'

/**
 * Meals component displays images of food
 */
const Meals = ({ mealItems }: MealsType) => (
  <section className="meals">
    {mealItems.map((meal) =>
      <figure key={meal.id} className="meal__photo">
        <img src={meal.imageURL} alt={meal.alt} />
      </figure>)}
  </section>
)

Meals.propTypes = {
  /**
   * Array contains 8 pictures of food
   */
  mealItems: CustomPropTypes.array(8).isRequired
}

export default Meals
