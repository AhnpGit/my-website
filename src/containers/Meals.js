// @flow

import { connect } from 'react-redux'
import Meals from '../components/Meals/Meals'
import type { MealsType } from '../utils/types'

type MealsState = {
  meals: MealsType
}

export const mapStateToProps = (state : MealsState) => ({
  mealItems: state.meals.mealItems
})

export default connect(
  mapStateToProps
)(Meals)
