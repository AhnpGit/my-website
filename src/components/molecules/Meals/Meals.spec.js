import React from 'react'
import Meals from './Meals'
import { shallow } from 'enzyme'

describe('Meals component', () => {
  it('renders correctly without issues', () => {
    const mealItems = [
      {
        id: 1,
        imageURL: "imageURL 1",
        alt: "alt 1"
      },
      {
        id: 2,
        imageURL: "imageURL 2",
        alt: "alt 2"
      },
      {
        id: 3,
        imageURL: "imageURL 3",
        alt: "alt 3"
      },
      {
        id: 4,
        imageURL: "imageURL 4",
        alt: "alt 4"
      },
      {
        id: 5,
        imageURL: "imageURL 5",
        alt: "alt 5"
      },
      {
        id: 4,
        imageURL: "imageURL 6",
        alt: "alt 4"
      },
      {
        id: 4,
        imageURL: "imageURL 7",
        alt: "alt 4"
      },
      {
        id: 4,
        imageURL: "imageURL 8",
        alt: "alt 4"
      },
    ]

    const result = shallow(
      <Meals mealItems={mealItems} />
    )

    expect(result).toMatchSnapshot()
  })
})
