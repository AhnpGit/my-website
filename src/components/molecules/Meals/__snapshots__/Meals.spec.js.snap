// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meals component renders correctly without issues 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section
    className="meals"
>
    <figure
        className="meal__photo"
    >
        <img
            alt="alt 1"
            src="imageURL 1"
        />
    </figure>
    <figure
        className="meal__photo"
    >
        <img
            alt="alt 2"
            src="imageURL 2"
        />
    </figure>
    <figure
        className="meal__photo"
    >
        <img
            alt="alt 3"
            src="imageURL 3"
        />
    </figure>
    <figure
        className="meal__photo"
    >
        <img
            alt="alt 4"
            src="imageURL 4"
        />
    </figure>
    <figure
        className="meal__photo"
    >
        <img
            alt="alt 5"
            src="imageURL 5"
        />
    </figure>
    <figure
        className="meal__photo"
    >
        <img
            alt="alt 4"
            src="imageURL 6"
        />
    </figure>
    <figure
        className="meal__photo"
    >
        <img
            alt="alt 4"
            src="imageURL 7"
        />
    </figure>
    <figure
        className="meal__photo"
    >
        <img
            alt="alt 4"
            src="imageURL 8"
        />
    </figure>
</section>,
  "nodes": Array [
    <section
      className="meals"
>
      <figure
            className="meal__photo"
      >
            <img
                  alt="alt 1"
                  src="imageURL 1"
            />
      </figure>
      <figure
            className="meal__photo"
      >
            <img
                  alt="alt 2"
                  src="imageURL 2"
            />
      </figure>
      <figure
            className="meal__photo"
      >
            <img
                  alt="alt 3"
                  src="imageURL 3"
            />
      </figure>
      <figure
            className="meal__photo"
      >
            <img
                  alt="alt 4"
                  src="imageURL 4"
            />
      </figure>
      <figure
            className="meal__photo"
      >
            <img
                  alt="alt 5"
                  src="imageURL 5"
            />
      </figure>
      <figure
            className="meal__photo"
      >
            <img
                  alt="alt 4"
                  src="imageURL 6"
            />
      </figure>
      <figure
            className="meal__photo"
      >
            <img
                  alt="alt 4"
                  src="imageURL 7"
            />
      </figure>
      <figure
            className="meal__photo"
      >
            <img
                  alt="alt 4"
                  src="imageURL 8"
            />
      </figure>
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Meals
        mealItems={
                Array [
                        Object {
                          "alt": "alt 1",
                          "id": 1,
                          "imageURL": "imageURL 1",
                        },
                        Object {
                          "alt": "alt 2",
                          "id": 2,
                          "imageURL": "imageURL 2",
                        },
                        Object {
                          "alt": "alt 3",
                          "id": 3,
                          "imageURL": "imageURL 3",
                        },
                        Object {
                          "alt": "alt 4",
                          "id": 4,
                          "imageURL": "imageURL 4",
                        },
                        Object {
                          "alt": "alt 5",
                          "id": 5,
                          "imageURL": "imageURL 5",
                        },
                        Object {
                          "alt": "alt 4",
                          "id": 4,
                          "imageURL": "imageURL 6",
                        },
                        Object {
                          "alt": "alt 4",
                          "id": 4,
                          "imageURL": "imageURL 7",
                        },
                        Object {
                          "alt": "alt 4",
                          "id": 4,
                          "imageURL": "imageURL 8",
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "mealItems": Array [
            Object {
              "alt": "alt 1",
              "id": 1,
              "imageURL": "imageURL 1",
            },
            Object {
              "alt": "alt 2",
              "id": 2,
              "imageURL": "imageURL 2",
            },
            Object {
              "alt": "alt 3",
              "id": 3,
              "imageURL": "imageURL 3",
            },
            Object {
              "alt": "alt 4",
              "id": 4,
              "imageURL": "imageURL 4",
            },
            Object {
              "alt": "alt 5",
              "id": 5,
              "imageURL": "imageURL 5",
            },
            Object {
              "alt": "alt 4",
              "id": 4,
              "imageURL": "imageURL 6",
            },
            Object {
              "alt": "alt 4",
              "id": 4,
              "imageURL": "imageURL 7",
            },
            Object {
              "alt": "alt 4",
              "id": 4,
              "imageURL": "imageURL 8",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section
          className="meals"
>
          <figure
                    className="meal__photo"
          >
                    <img
                              alt="alt 1"
                              src="imageURL 1"
                    />
          </figure>
          <figure
                    className="meal__photo"
          >
                    <img
                              alt="alt 2"
                              src="imageURL 2"
                    />
          </figure>
          <figure
                    className="meal__photo"
          >
                    <img
                              alt="alt 3"
                              src="imageURL 3"
                    />
          </figure>
          <figure
                    className="meal__photo"
          >
                    <img
                              alt="alt 4"
                              src="imageURL 4"
                    />
          </figure>
          <figure
                    className="meal__photo"
          >
                    <img
                              alt="alt 5"
                              src="imageURL 5"
                    />
          </figure>
          <figure
                    className="meal__photo"
          >
                    <img
                              alt="alt 4"
                              src="imageURL 6"
                    />
          </figure>
          <figure
                    className="meal__photo"
          >
                    <img
                              alt="alt 4"
                              src="imageURL 7"
                    />
          </figure>
          <figure
                    className="meal__photo"
          >
                    <img
                              alt="alt 4"
                              src="imageURL 8"
                    />
          </figure>
</section>,
        "_debugID": 2,
        "_renderedOutput": <section
          className="meals"
>
          <figure
                    className="meal__photo"
          >
                    <img
                              alt="alt 1"
                              src="imageURL 1"
                    />
          </figure>
          <figure
                    className="meal__photo"
          >
                    <img
                              alt="alt 2"
                              src="imageURL 2"
                    />
          </figure>
          <figure
                    className="meal__photo"
          >
                    <img
                              alt="alt 3"
                              src="imageURL 3"
                    />
          </figure>
          <figure
                    className="meal__photo"
          >
                    <img
                              alt="alt 4"
                              src="imageURL 4"
                    />
          </figure>
          <figure
                    className="meal__photo"
          >
                    <img
                              alt="alt 5"
                              src="imageURL 5"
                    />
          </figure>
          <figure
                    className="meal__photo"
          >
                    <img
                              alt="alt 4"
                              src="imageURL 6"
                    />
          </figure>
          <figure
                    className="meal__photo"
          >
                    <img
                              alt="alt 4"
                              src="imageURL 7"
                    />
          </figure>
          <figure
                    className="meal__photo"
          >
                    <img
                              alt="alt 4"
                              src="imageURL 8"
                    />
          </figure>
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Meals
    mealItems={
        Array [
            Object {
              "alt": "alt 1",
              "id": 1,
              "imageURL": "imageURL 1",
            },
            Object {
              "alt": "alt 2",
              "id": 2,
              "imageURL": "imageURL 2",
            },
            Object {
              "alt": "alt 3",
              "id": 3,
              "imageURL": "imageURL 3",
            },
            Object {
              "alt": "alt 4",
              "id": 4,
              "imageURL": "imageURL 4",
            },
            Object {
              "alt": "alt 5",
              "id": 5,
              "imageURL": "imageURL 5",
            },
            Object {
              "alt": "alt 4",
              "id": 4,
              "imageURL": "imageURL 6",
            },
            Object {
              "alt": "alt 4",
              "id": 4,
              "imageURL": "imageURL 7",
            },
            Object {
              "alt": "alt 4",
              "id": 4,
              "imageURL": "imageURL 8",
            },
          ]
    }
/>,
}
`;
